install_config/aggregate_logging_sizing.adoc

```diff
@@ -0,0 +1,255 @@
+logging-es-xmwvnorv-1-adr5x   1/1       Running     0          3d        ip-172-31-24-233.us-west-2.compute.internal
+logging-fluentd-156xn         1/1       Running     0          3d        ip-172-31-24-241.us-west-2.compute.internal
+logging-fluentd-40biz         1/1       Running     0          3d        ip-172-31-24-236.us-west-2.compute.internal
+logging-fluentd-7dtom         0/1       Pending     0          2d        ip-172-31-24-243.us-west-2.compute.internal
+logging-fluentd-8k847         1/1       Running     0          3d        ip-172-31-24-237.us-west-2.compute.internal
+logging-fluentd-9a3qx         1/1       Running     0          3d        ip-172-31-24-231.us-west-2.compute.internal
+logging-fluentd-abvgj         1/1       Running     0          3d        ip-172-31-24-228.us-west-2.compute.internal
+logging-fluentd-bh74n         1/1       Running     0          3d        ip-172-31-24-238.us-west-2.compute.internal
+...
+...
+
+----
+
+Notice how the pods are placed in different cluster nodes. +
+
+
+By default the amount of RAM allocated to each ES instance is 8GB. `*ES_INSTANCE_RAM*` is the parameter used in the openshift-ansible link:https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_examples/files/examples/v1.3/infrastructure-templates/enterprise/logging-deployer.yaml[template]. Keep in mind that *half* of this value will be passed to the individual elasticsearch pods java processes link:https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html#_give_half_your_memory_to_lucene[heap size].
+
+Further information regarding installing EFK can be found in the openshift-docs repository xref:../install_config/aggregate_logging.adoc[documentation]. 
+
+[[install-config-aggregate-logging-sizing-guidelines-rate-limiting]]
+== Systemd-journald and rsyslog 
+
+*Rate-limiting*
+
+In RHEL7 the `systemd-journald.socket` unit creates `/dev/log` during the boot process, and then passes input to systemd-journald.service. Every `syslog()` call goes to the journal. +
+Rsyslog uses the `imjournal` module as a default input mode for journal files. Refer to link:https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html-single/System_Administrators_Guide/index.html#s1-interaction_of_rsyslog_and_journal[Interaction of rsyslog and journal] for detailed information about this topic.
+
+
+At Red Hat, we have used a simple test harness which uses link:http://linux.die.net/man/1/logger[logger] across the cluster nodes to make entries of different sizes at different rates in the system log. During testing simulations under a default RHEL7 installation at certain logging rates ( ~ 40 log lines per second ), we've hit the default rate limit of `rsyslogd`.
+
+*Scaling up* +
+
+As you scale up your project, the default logging environment might need some adjustments.
+After upgrading to `systemd-219-22.el7.x86_64`, we've added:
```

Comment:
On Tue, Aug 30, 2016 at 12:02 PM, Ricardo Lourenço <notifications@github.com
> wrote:

> In install_config/aggregate_logging_sizing.adoc
> <https://github.com/openshift/openshift-docs/pull/2723#discussion_r76825728>
> :
>
> > +
> > +[[install-config-aggregate-logging-sizing-guidelines-rate-limiting]]
> > +== Systemd-journald and rsyslog
> > +
> > +*Rate-limiting*
> > +
> > +In RHEL7 the `systemd-journald.socket` unit creates `/dev/log` during the boot process, and then passes input to systemd-journald.service. Every `syslog()` call goes to the journal. +
> > +Rsyslog uses the `imjournal` module as a default input mode for journal files. Refer to link:https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html-single/System_Administrators_Guide/index.html#s1-interaction_of_rsyslog_and_journal[Interaction of rsyslog and journal] for detailed information about this topic.
> > +
> > +
> > +At Red Hat, we have used a simple test harness which uses link:http://linux.die.net/man/1/logger[logger] across the cluster nodes to make entries of different sizes at different rates in the system log. During testing simulations under a default RHEL7 installation at certain logging rates ( ~ 40 log lines per second ), we've hit the default rate limit of `rsyslogd`.
> > +
> > +*Scaling up* +
> > +
> > +As you scale up your project, the default logging environment might need some adjustments.
> > +After upgrading to `systemd-219-22.el7.x86_64`, we've added:
>
> How about:
>
After this, we have noticed that entries stopped being written to journald
> due to local journal file corruption, verifiable with journalctl --verify.
> Restarting the logging daemon does not fix the issue so it was necessary
> to update systemd.
>

​No, I think be more clear about per my original comment. Point to the
specific errata.




​




> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/2723/files/a9344c708501203429adeafb1e3ab47484e481c4#r76825728>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1KmGDwVfyrE6rGpDql5IRNqco--bgEks5qlFP-gaJpZM4JsARl>
> .
>



-- 

-- Jeremy Eder